CMakeLists.txt

```diff
@@ -0,0 +1,138 @@
+cmake_minimum_required(VERSION 3.16)
+project(wfa2lib)
+
+set(CMAKE_CXX_STANDARD 17)
+
+include(FeatureSummary)
+
+find_package(PkgConfig REQUIRED)
+
+feature_summary(
+  FATAL_ON_MISSING_REQUIRED_PACKAGES
+  WHAT REQUIRED_PACKAGES_NOT_FOUND)
+
+# ---- Options
+
+option(OPENMP "Enable OpenMP" ON)
+option(PROFILING "Enable profiling" OFF)
+option(ASAN "Use address sanitiser" OFF)
+
+# include(CheckIPOSupported) # adds lto
+# check_ipo_supported(RESULT ipo_supported OUTPUT output)
+
+# ---- Dependencies
+
+if(OPENMP)
+  include(FindOpenMP)
+endif(OPENMP)
+
+find_package(Threads)
+set_package_properties(Threads PROPERTIES TYPE REQUIRED)
+
+# ---- Build switches
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+# set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ${ipo_supported})
+
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE Release CACHE STRING
+          "Choose the type of build, options are: Release|Debug|RelWithDebInfo (for distros)." FORCE)
+endif()
+
+if (${CMAKE_BUILD_TYPE} MATCHES Release)
+  set(EXTRA_FLAGS "-march=native -D_FILE_OFFSET_BITS=64")
+endif()
+
+if ((${CMAKE_BUILD_TYPE} MATCHES Release) OR (${CMAKE_BUILD_TYPE} MATCHES RelWithDebInfo))
```

Comment:
@pjotrp I would suggest not using the WFA_PARALLEL flag in 'Release' mode, as it will make the whole WFA2-lib a bit slower.